enderio-base/src/main/java/com/enderio/base/common/item/tool/PoweredToggledItem.java

```diff
@@ -0,0 +1,106 @@
+import com.enderio.core.common.capability.MultiCapabilityProvider;
+import com.enderio.core.common.util.EnergyUtil;
+import net.minecraft.core.NonNullList;
+import net.minecraft.nbt.CompoundTag;
+import net.minecraft.world.InteractionHand;
+import net.minecraft.world.InteractionResultHolder;
+import net.minecraft.world.entity.Entity;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.item.CreativeModeTab;
+import net.minecraft.world.item.Item;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.level.Level;
+import net.minecraftforge.common.util.LazyOptional;
+import net.minecraftforge.energy.CapabilityEnergy;
+import net.minecraftforge.energy.EnergyStorage;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+public abstract class PoweredToggledItem extends Item implements IEnergyBar, IMultiCapabilityItem {
+
+    public PoweredToggledItem(Properties pProperties) {
+        super(pProperties.stacksTo(1));
+    }
+
+    protected abstract void onTickWhenActive(Player player, @Nonnull ItemStack pStack, @Nonnull Level pLevel, @Nonnull Entity pEntity, int pSlotId, boolean pIsSelected);
+
+    protected abstract int getEnergyUse();
+
+    protected abstract int getMaxEnergy();
+
+    protected void enable(ItemStack stack) {
+        Toggled.setEnabled(stack, true);
+    }
+
+    protected void disable(ItemStack stack) {
+        Toggled.setEnabled(stack, false);
+    }
+
+    @Override
+    public boolean isFoil(ItemStack pStack) {
+        return Toggled.isEnabled(pStack);
+    }
+
+    @Override
+    public void fillItemCategory(@Nonnull CreativeModeTab pCategory, @Nonnull NonNullList<ItemStack> pItems) {
+        if (allowdedIn(pCategory)) {
+            ItemStack is = new ItemStack(this);
+            pItems.add(is.copy());
+
+            EnergyUtil.setFull(is);
+            pItems.add(is);
+        }
+    }
+
+    @Nonnull
+    @Override
+    public InteractionResultHolder<ItemStack> use(@Nonnull Level pLevel, Player pPlayer, @Nonnull InteractionHand pUsedHand) {
+        if (pPlayer.isCrouching()) {
+            ItemStack stack = pPlayer.getItemInHand(pUsedHand);
+            if(Toggled.isEnabled(stack)) {
+                disable(stack);
+            } else if (EnergyUtil.extractEnergy(stack, getEnergyUse(), true) > 0) {
+                enable(stack);
+            }
+        }
+        return super.use(pLevel, pPlayer, pUsedHand);
+    }
+
+    @Override
+    public void inventoryTick(@Nonnull ItemStack pStack, @Nonnull Level pLevel, @Nonnull Entity pEntity, int pSlotId, boolean pIsSelected) {
+        if (pEntity instanceof Player player) {
+            if (Toggled.isEnabled(pStack)) {
+                if (EnergyUtil.extractEnergy(pStack, getEnergyUse(), true) > 0) {
+                    EnergyUtil.extractEnergy(pStack, getEnergyUse(), false);
+                    onTickWhenActive(player, pStack,pLevel,pEntity,pSlotId, pIsSelected);
+                } else {
```

Comment:
add an empty method like stillValid(ItemStack pStack) for the Leviation Staff to return false if the fluid tank is empty, so that the item is disabled like with the EnergyUtils.extracEnergy check.
Even though that method is not implemented in Leviation Staff as it doesn't have a fluid container right now, the method can be added now